confirm.c

```diff
@@ -780,19 +878,64 @@ confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
 	query->ttl = ttl;
 	query->traffic_class = traffic_class;
 	query->flow_label = flow_label;
-	query->icmpid = icmpid;
+	query->src_port = src_port;
+	query->dst_port = dst_port;
+	query->icmpid = 0;
+	query->flowid = flowid & CONFIRM_MAX_FLOWID;
+	query->padding = 0;
+	query->revflow = 0;
+	query->ntries = 3;
+	query->cb = cb;
+	query->data = NULL;
+	query->trynum = 0;
+	query->probetime.tv_sec = 2;
+	query->probetime.tv_nsec = 0;
+	query->timeout.tv_sec = 5;
+	query->timeout.tv_nsec = 0;
+	query->start.tv_sec = 0;
+	query->start.tv_nsec = 0;
+	query->lastpkt.tv_sec = 0;
+	query->lastpkt.tv_nsec = 0;
+	query->answertime.tv_sec = 0;
+	query->answertime.tv_nsec = 0;
+	query->event = NULL;
+	query->probe = NULL;
+	query->response = NULL;
+	query->type = PROBE_TYPE_TCP;
+	return query;
+}
+
+struct confirm_query *
+confirm_query_create6_icmp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint16_t icmpid,
+		uint8_t flowid, confirm_query_cb cb)
+{
+	struct confirm_query *query = malloc(sizeof(struct confirm_query));
+	
+	if(!query){
+		logea(__FILE__, __LINE__, NULL);
+	}
+
 	if(flowid > CONFIRM_MAX_FLOWID) {
 		logd(LOG_WARN, "%s,%d: flowid > 127!\n", __FILE__, __LINE__);
 	}
+
+	memcpy(&(query->dst), dst, sizeof(query->dst));
+	memset(&(query->ip), UINT8_MAX, sizeof(query->ip));
+	query->ip.ss_family = dst->ss_family;
+	query->ttl = ttl;
+	query->traffic_class = traffic_class;
+	query->flow_label = flow_label;
+	query->src_port = 0;
```

Comment:
this seems enough for now. in the future, we might want to pack the try number (see `query->ntries`) in the probe (currently, when we retry a probe and send it again because of timeouts, we cannot identify which try was answered)